.github/workflows/ci.yaml

```diff
@@ -620,6 +620,51 @@ jobs:
           python --version
           pylint --ignore-missing-annotations=y homeassistant/components/${{ needs.info.outputs.integrations_glob }}
 
+  pylint-tests:
+    name: Check pylint on tests
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    if: |
+      github.event.inputs.mypy-only != 'true'
+      || github.event.inputs.pylint-only == 'true'
+    needs:
+      - info
+      - base
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v4.1.6
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        id: python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+          check-latest: true
+      - name: Restore full Python ${{ env.DEFAULT_PYTHON }} virtual environment
+        id: cache-venv
+        uses: actions/cache/restore@v4.0.2
+        with:
+          path: venv
+          fail-on-cache-miss: true
+          key: >-
+            ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{
+            needs.info.outputs.python_cache_key }}
+      - name: Register pylint problem matcher
+        run: |
+          echo "::add-matcher::.github/workflows/matchers/pylint.json"
+      - name: Run pylint (fully)
+        if: needs.info.outputs.test_full_suite == 'true'
+        run: |
+          . venv/bin/activate
+          python --version
+          pylint --jobs=1 --ignore-missing-annotations=y tests
+      - name: Run pylint (partially)
+        if: needs.info.outputs.test_full_suite == 'false'
+        shell: bash
+        run: |
+          . venv/bin/activate
+          python --version
+          pylint --jobs=1 --ignore-missing-annotations=y tests/components/${{ needs.info.outputs.integrations_glob }}
```

Comment:
Why are we setting `--jobs=1`

Reply:
This is because of an issue with `per-file-ignores`
See https://github.com/christopherpickering/pylint-per-file-ignores/issues/101

I haven't tried it recently though - so maybe worth a try